crates/solver/src/settlement_ranker.rs

```diff
@@ -200,17 +192,90 @@ impl SettlementRanker {
                 )),
             );
         }
-        for (i, (solver, _, _)) in enumerate(&rated_settlements) {
-            let rank = rated_settlements.len() - i;
-            solver.notify_auction_result(auction_id, AuctionResult::Ranked(rank));
-            self.metrics
-                .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
-        }
 
+        // Before sorting, make sure to shuffle the settlements. This is to make sure we
+        // don't give preference to any specific solver when there is an
+        // objective value tie.
+        rated_settlements.shuffle(&mut rand::thread_rng());
+
+        if cfg!(feature = "auction-rewards") {
```

Comment:
I'm not a big fan of `cfg!(feature)` here. Mostly because I think this should be a runtime and not a compile time flag.

Reply:
Switched to compile time check.

Reply:
It's still a compile-time check at the moment with `#[cfg(feature = "auction-rewards")]` . Maybe didn't commit?

Reply:
Or maybe misunderstanding. Nick is saying (and I agree) that this should *not* be a compile time check. It should be a normal configuration option.

Reply:
I thought Nick confused and wanted to say the opposite (but i forgot to comment/ask here). 
Originally I wanted to use the feature flag but ended up using the flag only at 1 place in the code, so turning on/off with configuration parameter here makes sense. 
Added.

Reply:
This is what I meant, I think it should be a configuration flag and _not_ a compile time flag.